it/src/test/java/apoc/it/core/RBACOnLoadTest.java

```diff
@@ -0,0 +1,158 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package apoc.it.core;
+
+import static apoc.util.MapUtil.map;
+import static apoc.util.TestContainerUtil.createEnterpriseDB;
+import static apoc.util.TestContainerUtil.testCall;
+import static apoc.util.TestContainerUtil.testCallEmpty;
+import static java.util.Collections.emptyMap;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.assertTrue;
+
+import apoc.util.Neo4jContainerExtension;
+import apoc.util.TestContainerUtil;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.UnknownHostException;
+import java.util.List;
+import java.util.Map;
+import org.assertj.core.api.Assertions;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.neo4j.driver.AuthTokens;
+import org.neo4j.driver.Driver;
+import org.neo4j.driver.GraphDatabase;
+import org.neo4j.driver.Session;
+
+public class RBACOnLoadTest {
+
+    private static Neo4jContainerExtension neo4jContainer;
+    private static Driver testUserDriver;
+    private static Session session;
+    private static Session testUserSession;
+    private static final String user = "testUser";
+    private static final String userP = "password1234";
+
+    @BeforeClass
+    public static void beforeClass() {
+        neo4jContainer = createEnterpriseDB(List.of(TestContainerUtil.ApocPackage.CORE), true)
+                .withNeo4jConfig("dbms.memory.heap.max_size", "1GB");
+        neo4jContainer.start();
+
+        assertTrue(neo4jContainer.isRunning());
+        session = neo4jContainer.getSession();
+
+        setupUser();
+        testUserDriver = GraphDatabase.driver(neo4jContainer.getBoltUrl(), AuthTokens.basic(user, userP));
+        testUserDriver.verifyConnectivity();
+        testUserSession = testUserDriver.session();
+    }
+
+    private static void setupUser() {
+        List<String> queries = List.of(
+                "CREATE ROLE test",
+                "CREATE USER " + user + " SET PASSWORD '" + userP + "' SET PASSWORD CHANGE NOT REQUIRED",
+                "GRANT ROLE test, reader, editor TO testUser");
+        for (String query : queries) testCallEmpty(session, query, emptyMap());
+    }
+
+    private static void addRBACOnLoad(String urlString) throws MalformedURLException, UnknownHostException {
+        InetAddress address = InetAddress.getByName(new URL(urlString).getHost());
+        String ip = address.getHostAddress();
+        String query = "DENY LOAD ON CIDR \"" + ip + "/32\" TO test";
+
+        testCallEmpty(session, query, emptyMap());
+    }
+
+    @Test
+    public void testRBACOnDeny() throws IOException {
+        String url = "https://neo4j.com/docs/cypher-refcard/3.3/csv/artists.csv";
+        addRBACOnLoad(url);
+
+        List<String> loadableAPOCProcs = List.of(
```

Comment:
I looked at the docs for apoc.load.* and apoc.import.* to compare with this list, and there also seems to be apoc.load.arrow.stream

Reply:
apoc.load.arrow.stream takes a byte array and not a url :)